runtime-transaction/src/runtime_transaction.rs

```diff
@@ -0,0 +1,164 @@
+use {
+    crate::{
+        simple_vote_transaction_checker::is_simple_vote_transaction, transaction_meta::TransactionMeta,
+    },
+    solana_sdk::{
+        hash::Hash,
+        message::{
+            AddressLoader, SanitizedMessage, SanitizedVersionedMessage,
+        },
+        signature::Signature,
+        transaction::{Result, SanitizedVersionedTransaction},
+    },
+};
+
+/// RuntimeTransaction is runtime face representation of transaction, as
+/// SanitizedTransaction is client facing rep.
+/// The lifetime of RuntimeTransaction within banking_stage is roughly this: 
+/// 1. Receive `packet` from sigverify, deserialize it into i`solana_sdk::VersioedTransaction`
+/// 2. sanitize into `solana_sdk::SanitizedVersionedTransaction`
+/// 3. create `RuntimeTransaction` from it, which does:
+///    3.1 sets `RuntimeMessage` to `StaticallyLoaded` state
+///    3.2 extracts static meta: message hash, simple vote flag, compute budget limits, payer account etc
+///        Perhaps evencalculates transaction cost at this point (need upgrade cost-model a bit)
+/// 4. banking_stage/scheduler process `RuntimeTransaction` with its static metadata, untill when
+///    it becomes necessary to call `load_addresses(...)` load accounts address from on-chain ALT
+///    4.1 `RuntimeMessage` transits to `DynamicallyLoaded` state
+///    4.2 extracts dynamic metadata, such as nonce account etc
+/// 5. `RuntimeTransaction` is ready for "load_and_execute"
+///
+
+/// runtime message has two stats:
+/// 1. Initial state when message
+/// is created from solana_sdk::SanitizedVersionedTransaction,
+/// which is a super-light-wieghted op; During this state,
+/// static metadata (eg those solely based on message itself)
+/// can be cheaply resolved and used.
+/// 2. DynamicallyLoaded state is when account addresses loaded from on-chain
+/// address-lookup-table, a operation involves loading accounts from
+/// accounts db, deserialize accounts' data. During this state,
+/// accounts related metadata can be resolved, such as Nonce account,
+/// transaction costs, and transaction (base) fee
+///
+/// Following events can transit runtime message from Initial state to DynamicallyLoaded state:
+/// 1. `pub fn load_addresses(address_loader)` is called, or
+///
+/// New Error:
+/// If Getter of account-based metadata is called while in Initial state,
+/// a new Error shall return to indicate RuntimeTransaction is not
+/// furnished with necessary data yet.
+///
+#[derive(Debug, Default, Clone, Eq, PartialEq)]
+pub enum RuntimeMessage {
+    #[default]
+    Uninitialized,
+    StaticallyLoaded(SanitizedVersionedMessage),
+    DynamicallyLoaded(SanitizedMessage),
```

Comment:
enum variants VS individual structs, can't decide. 

with enum, i have one `RuntimeTransaction` that can be in 3 states; But call site needs to `match ...` all the time?

Or can have three separate structs: `UninitializedRuntimeTransaction`,`StaticallyLoadedRuntimeTransaction` and `DynamicallyLoadedRuntimeTransaction`. Awful names, but strong typed, so cleaner. One question is if a function takes `StaticallyLoadedRuntimeTransaction`, but you have `DynamicallyLoadedRuntimeTransaction`, how do you downcast?

Reply:
Dealing with the "downcasting" probably has 2 approaches:

1. w/ separate structs - each struct implements some trait to access some info
2. w/ enums - top-level fn, match & access